addons/marketing_card/models/mailing_mailing.py

```diff
@@ -0,0 +1,72 @@
+from odoo import _, api, exceptions, fields, models, osv
+
+
+class MassMailing(models.Model):
+    _name = 'mailing.mailing'
+    _inherit = 'mailing.mailing'
+
+    mailing_model_id = fields.Many2one(compute="_compute_mailing_model_id", store=True, readonly=False)
+    card_requires_sync_count = fields.Integer(compute="_compute_card_requires_sync_count")
+    card_campaign_id = fields.Many2one('card.campaign')
+
+    @api.constrains('card_campaign_id', 'mailing_domain', 'mailing_model_id')
+    def _check_mailing_domain(self):
+        for mailing in self:
+            if mailing.card_campaign_id:
+                if mailing.sudo().mailing_model_id.model != mailing.card_campaign_id.res_model:
+                    raise exceptions.ValidationError(_(
+                        "Card Campaign Mailing should target model %(model_name)s",
+                        model_name=self.env['ir.model']._get(mailing.card_campaign_id.res_model).display_name
+                    ))
+
+    @api.depends('card_campaign_id')
+    def _compute_mailing_model_id(self):
+        for mailing in self.filtered('card_campaign_id'):
+            mailing.mailing_model_id = self.env['ir.model']._get_id(mailing.card_campaign_id.res_model)
+
+    @api.depends('card_campaign_id')
+    def _compute_card_requires_sync_count(self):
+        """Check if there's any missing or outdated card."""
+        self.card_requires_sync_count = 0
+        # no point in updating sent mailings
+        card_mailings = self.filtered(lambda mailing: mailing.card_campaign_id and mailing.state == 'draft')
+        for mailing in card_mailings:
+            TargetModel = self.env[mailing.mailing_model_real]
+            expected = TargetModel.search(self._card_parse_mailing_domain_raw())
+            actual = TargetModel.search(self._parse_mailing_domain())
+            missing_count = len(expected) - len(actual) 
+            out_of_date_count = self.env['card.card'].search_count([
+                ('campaign_id', '=', mailing.card_campaign_id.id),
+                ('res_id', 'in', expected.ids),
+                ('generation', '<', mailing.card_campaign_id.generation)
+            ])
+            mailing.card_requires_sync_count = missing_count + out_of_date_count
```

Comment:
Can't we have something simpler (without hacking parse mailing domain), like
```suggestion
            recipients = self.env[mailing.mailing_model_real].search(self._parse_mailing_domain())
            valid_cards_count = self.env["card.card"].search_count([
                ("campaign_id", "=", mailing.card_campaign_id.id),
                ("id", "in", recipients.ids),
                ("outdated", "=", False),
            ])
            mailing.card_requires_sync_count = len(recipients) - valid_cards_count
```